servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -287,14 +302,40 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge)
           .merge(
               toHash(merge.getFromRefName(), merge.getFromHash(), true).get(),
               BranchName.of(branchName),
-              toHash(hash, true));
+              toHash(hash, true),
+              getResetMergePropsFunc(MergeOp.MERGE, merge.getFromRefName()));
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(e.getMessage(), e);
     } catch (ReferenceConflictException e) {
       throw new NessieConflictException(e.getMessage(), e);
     }
   }
 
+  private BiFunction<Serializer<CommitMeta>, ByteString, ByteString> getResetMergePropsFunc(
+      MergeOp mergeOp, String fromRefName) {
+    return (ser, in) -> {
+      try {
+        CommitMeta commitMeta = ser.fromBytes(in);
+        String currUser =
+            Optional.ofNullable(getSecurityContext().getUserPrincipal())
+                .map(Principal::getName)
+                .orElse("");
+        String mergeProp = String.format("%s_ref", mergeOp.name().toLowerCase());
+        CommitMeta mergeReadyMeta =
+            CommitMeta.builder()
+                .from(commitMeta)
+                .committer(currUser)
+                .commitTime(Instant.now())
+                .putProperties(mergeProp, fromRefName)
+                .build();
+        return ser.toBytes(mergeReadyMeta);
+      } catch (RuntimeException e) { // error in serialization
```

Comment:
This should either fail completely or succeed, but only logging a warning isn't helpful IMO

Reply:
Is this mapping of properties considered as critical to fail the merge/translate operation?

Just want to confirm before I make that change.

Reply:
Either it fails or succeeds. But failing with a logged warning that probably nobody looks at is not good IMO.
If serialization fails, then something is very broken, right?